### PR TITLE
Branch asssertion and logging

### PR DESCRIPTION
**Pull request by ivanthengwr**
Added in loggers, assertions and more exceptions to the code to make it more defensive. I have also updated the text-ui-test inputs.

### CODE REVIEW

File: src/main/java/seedu/duke/SkyControl.java [hunk 1]
```diff
@@ -40,8 +44,16 @@ private void checkEntity(String lineInput) {
         isFlight = Parser.isFlightEntity(lineInput);
     }
 
+    private void setUpAllLogger() {
```
Comment (Franky4566) [inline src/main/java/seedu/duke/SkyControl.java:47]: Good use of SLAP, easy for other teammates to add on loggers from here too

File: src/main/java/seedu/duke/parsers/PassengerParser.java [hunk 2]
```diff
@@ -17,7 +17,7 @@ public static Command parse(String[] inputWords) throws SkyControlException {
         } else if (isList) {
             command = new ListPassengerCommand();
         } else {
-            throw new SkyControlException(ui.showErrorMessage());
+            throw new SkyControlException(ui.getErrorMessage());
```
Comment (Franky4566) [inline src/main/java/seedu/duke/parsers/PassengerParser.java:20]: Good correction of method names to verbs to follow java coding standard

**Review by Franky4566 — COMMENTED** [review]
No negative comments, good use of slap for future expansion too.

File: src/main/java/seedu/duke/command/Command.java [hunk 3]
```diff
@@ -28,13 +24,13 @@ public static void getPassengerDetail(String lineInput) throws SkyControlExcepti
             checkBlankDetailInput();
             passengerDetail = passengerDetailArray[DETAIL_INDEX].trim();
         } else {
-            throw new SkyControlException(ui.showOperationError());
+            throw new SkyControlException(ui.getOperationError());
         }
     }
 
     public static void checkBlankDetailInput() throws SkyControlException {
         if (passengerDetailArray.length < 2) {
```
Comment (JordanKwua) [inline src/main/java/seedu/duke/command/Command.java:32]: Could use a constant to define this magic number to be clearer instead

**Review by JordanKwua — COMMENTED** [review]
A very good job overall! Separated the commands into individual classes to make it more OOP.

**Pull request merged**